packages/dashboard/src/data/listAssetPropertiesMap/fetchListAssetPropertiesMap.tsx

```diff
@@ -0,0 +1,32 @@
+import { IoTSiteWiseClient } from '@aws-sdk/client-iotsitewise';
+import {
+  SiteWiseAssetQuery,
+  SiteWisePropertyAliasQuery,
+} from '@iot-app-kit/source-iotsitewise';
+import { QueryClient } from '@tanstack/react-query';
+import {
+  createFetchSiteWiseAssetQueryDescription,
+  createListAssetPropertiesMapCacheKey,
+} from './query';
+import { selectListAssetPropertiesMap } from './selectData';
+
+export const fetchListAssetPropertiesMap = async (
+  siteWiseQuery:
+    | Partial<SiteWiseAssetQuery & SiteWisePropertyAliasQuery>
+    | undefined,
+  queryClient: QueryClient,
+  iotSiteWiseClient: IoTSiteWiseClient
+) => {
+  const queryKey = createListAssetPropertiesMapCacheKey(siteWiseQuery);
+  const queryFn = createFetchSiteWiseAssetQueryDescription(
+    iotSiteWiseClient,
+    siteWiseQuery
+  );
+
+  const data = await queryClient.fetchQuery({
+    queryKey,
+    queryFn,
+  });
+
+  return selectListAssetPropertiesMap(data);
+};
```

Comment:
This is exactly how we discussed. Glad to see everything you needed was available!